app_legacy/Helpers/database/game.php

```diff
@@ -148,13 +163,22 @@ function getGameMetadata(
             $requestedByStatement = 'OR ua.User = :username';
         }
 
+        $gameIdStatement = 'ach.GameID = :gameId';
+        if ($parentGameId !== null) {
+            $bindings['parentGameId'] = $parentGameId;
+            $gameIdStatement = 'ach.GameID IN (:gameId, :parentGameId)';
+        }
+
         $query = "SELECT aw.HardcoreMode, COUNT(DISTINCT aw.User) as Users
-                FROM Awarded AS aw
-                LEFT JOIN Achievements AS ach ON ach.ID = aw.AchievementID
-                LEFT JOIN UserAccounts as ua ON ua.User = aw.User
-                WHERE ach.GameID = :gameId AND ach.Flags = :achievementType
-              AND (NOT ua.Untracked $requestedByStatement)
-              GROUP BY aw.HardcoreMode";
+                FROM (
+                  SELECT aw.User, aw.HardcoreMode
+                  FROM Awarded AS aw
+                  LEFT JOIN Achievements AS ach ON ach.ID = aw.AchievementID
+                  LEFT JOIN UserAccounts AS ua ON ua.User = aw.User
+                  WHERE $gameIdStatement AND ach.Flags = :achievementType
+                  AND (NOT ua.Untracked $requestedByStatement)
+                ) AS aw
+                GROUP BY aw.HardcoreMode";
```

Comment:
There's something happening with the DW3(GBC) set:

Parent:
![image](https://user-images.githubusercontent.com/32680403/236570960-4bf308e4-4b4b-44e2-8cda-d0b51c7462fb.png)

Subset:
![image](https://user-images.githubusercontent.com/32680403/236571040-fd863b85-3d0b-4081-a0d1-c2f39bf2e20b.png)

Without digging into the data, I assume this is because the relationship is not reciprocal. A player who has only played the subset won't show up in the total number of players for the parent game query.

There probably isn't a good solution for this, as the parent would have to count players from all subsets, and that information would have to be propagated to all subsets.

Reply:
You are correct, the relationship is indeed not reciprocal. I came to the same conclusion that there wouldn't be a good solution for this.